### ISSUE TITLE
Research: Test: Issue Template: Update Wiki Page

### ISSUE THREAD

**Issue by jinyan0425** [hackforla/internship#611]
#### Instructions for creating this issue
The person who creates this issue should know which template draft needs to be tested. They will add links to the `Resources for creating this issue` section and customize this issue.
- [ ] Choose `Submit new issue` button at the bottom to create a new issue 
- Customize the issue
    - [ ] Select the three dots on the top right of the comment, and select "Edit"
    - [ ] Copy and paste entire template text in this issue into a text editor (ie., "TextEdit" on _Mac_, "Wordpad" on _PC_)
    - Use Find/Replace (**Mac:** ⌘+F or **PC**: Ctrl+F) to search for and replace the keyword strings (including text and the brackets) below with what the keyword or information is asking for (which will be in the create template draft issue this issue is testing)
        - [ ] Search for `[Replace with LINK TO TEMPLATE DRAFT ISSUE COMMENT]` 
        - [ ] Search for `[Replace with DEFAULT TITLE THAT ISSUE TEMPLATE WILL HAVE]`
        - [ ] Search for `[Replace DIRECT OBJECT]`
        - [ ] Search for `[Replace with NAME OF TEMPLATE]`
    - [ ] Copy and paste the edited template from the text editor back into this GitHub Issue, replacing all original text
    - [ ] Choose `Update comment` to save the changes
    - [ ] Re-check all the checkboxes above after copying and pasting the edited template back in this issue
    - [ ] Close the text editor
- Customize the issue title
    - [ ] Copy
    ```
    Research: Test: Issue Template: [Replace with NAME OF TEMPLATE]
    ```
    - [ ] Choose `Edit` to the right side of the title
    - [ ] Paste what you copied to replace the existing title
    - [ ] Choose `Save`
- Customize the issue settings (located in right sidebar)
    - [ ] Under Labels, add the feature that matches which template you are testing (e.g., if you were testing the roadmap template, you would choose `feature: roadmap`)
        - [ ] If you are not sure of the label(s) you are adding, please leave a new comment in this issue to inform Research Leads so they could double check to ensure the appropriate labels are included 
    - [ ] Add to the project board, by choosing the gear next to Projects, choose `Repository`, and choose `TWE: Project Board`
    - [ ] Add the milestone: `06.00 Research Guidance`
    - [ ] Remove the label `Milestone: missing`
- Customize dependencies 
    - [ ] If no dependencies, delete dependency section
    - [ ] If dependencies, add the dependencies (any existing issue or a reason that would prevent the action items from being completed)
- Customize the `Resources for creating this issue` section at the end of this comment
    - [ ] Locate the create template draft issue that needs to be tested
    - [ ] Locate the template draft in the comments of the issue (not the issue template in the first comment)
    - [ ] Copy the comment link of the template draft 
    - [ ] Add `()` at the end of Resources 1.01 of this new issue with no space after`]`
    - [ ] Paste the comment link you just copied inside the `()` at the end of Resources 1.01 so it becomes a hyperlink
    - [ ] Check the rest of the `Resources for creating this issue` section to make sure the wording make sense and the links work
- [ ] Add `status: template testing (step 2b)` label to the create template draft issue (which is the issue associated with Resources 1.01)
- Remove this section
    - [ ] Delete this section when all the checkboxes above are complete (i.e., the issue assignee will never see this part)
    - Depending on your role, please do the following:
       - [ ] Research Lead: put this issue into the prioritized backlog
       - [ ] Team member: add the `ready for research lead` label

### Dependencies
[Replace with the link to the issue template issue this issue will review, when its issue template is ready for testing]

### Overview
We need to test the [Replace DIRECT OBJECT] Issue Template so that researchers can start making [Replace DIRECT OBJECT] issues on/for the various research plans/tasks

#### Details
We expect that the template you are testing will need some improvement and the fact that you don't know much about the project is a benefit here, because it's likely that the person making the template can overlook something that they already know how to do, but should be fully explained in the template so that people can do the work independently.

### Action Items
- [ ] Go to the draft issue template in a comment on the issue where the draft you are testing is being created in Resource 1.01
- Copy the draft
   - [ ] Click on three dots
   - [ ] Choose edit 
   - [ ] Select the contents (Use Select All **Mac:** ⌘+A or **PC**: Ctrl+A) 
   - [ ] Copy the contents (Use Copy **Mac:** ⌘+C or **PC**: Ctrl+C) 
- Create the new issue
   - [ ] Open a new blank issue template (Resource 1.02)
- Use the template to customize the issue
   - [ ] Paste the draft into the new issue body
   - [ ] Replace the title with the following text
      ```
      `[TEST]` 
      ```
   - [ ] Add the title from Resource 1.03 after the text: `[TEST]`.  This part is the title that would have been provided if the issue was made from a template instead of this testing process. 
   - [ ] Submit the new issue
- Add Labels to the new issue
   - [ ] Locate the labels associated with the template to be tested (see Resource 1.01)
      <details><summary>Example: See the highlighted section</summary>
      <p>
      <img width="400" alt="Screenshot 2024-05-28 at 11 35 24 AM" src="https://github.com/hackforla/internship/assets/57029070/4146ecaa-a893-41e7-9c65-3c0704eb1686">
      <p>
      </details>               
   - Add all the labels associated with the template to be tested in the new issue you just submitted
      - [ ] Under Labels, choose gear <img src="https://user-images.githubusercontent.com/57029070/211904252-46521281-b8f8-495d-b31f-d06e425f0096.png" width=25px>
      - [ ] Select each label you previously located from the template to be tested
   - [ ] Add `status: template test (step 3)` label to the new issue you just submitted
- Add new issue to Project Board
   - [ ] Under Projects, choose gear <img src="https://user-images.githubusercontent.com/57029070/211904252-46521281-b8f8-495d-b31f-d06e425f0096.png" width=25px>, choose `Repository`, and choose `TWE: Project Board`
   - [ ] Under `TWE: Project Board`, choose the down arrow and change the status to `In Progress`
- Assign yourself the issue
   - [ ] Under Assignees, Choose No one - assign yourself
- Use the new issue to test the template
   - [ ] Follow the instructions in the new issue.  Your goal is to test that the template is sufficient for creating the issue and working on it, so for the purposes of this test, you are the creator of the issue and the assignee.
   - [ ] If you have any questions about the instructions in this issue, leave a new comment below. Then move this issue to the `Questions/Review` column, and add `ready for research lead` label
   - [ ] When you have question or suggestions about the instructions in the new issue you are using to test the template, please:
      - [ ] leave comments in that new issue, @ messaging the person who made the template in your comment
      - [ ] move it to the Questions/Review column
      - [ ] add `ready for research lead` label
      - [ ] @ message the person who made the template in Slack with a link to your comment.
   - [ ] Review changes to the template made by author and repeat your review process until you are satisfied that it's ready for deployment
      - [ ] Leave a comment on the template issue and the new issue you created with your sign-off (@ messaging both the template author and the Research Lead)
      - [ ] Add the issues to the next Research team agenda (Resource 1.04) and @ messaging both the template author and the Research Lead via the team Slack channel with a link to the issue comments)

#### Research lead action items
- [ ] review the template again
   - [ ] if changes are needed, send back to author with notes
   - [ ] if no changes are needed, deploy template, close issues.
      - [ ] Add link to the finished template to Resource 2.01

#### Resources for creating this issue
- 1.01 [Replace with LINK TO TEMPLATE DRAFT ISSUE COMMENT]
- 1.02 [Blank issue template](https://github.com/hackforla/internship/issues/new)
- 1.03 [Replace with DEFAULT TITLE THAT ISSUE TEMPLATE WILL HAVE]
- 1.04 https://github.com/hackforla/internship/issues/62

#### Resources gathered during the completion of this issue
- 2.01 [Replace with NAME OF TEMPLATE]

**Comment by jinyan0425** [comment]
@pandanista Ying, I have created this testing issue, however, it cannot be finished until we INDEED have a RP that needs wiki updates. A minor question on the labeling: shall we add "p-feature: wiki" for this template? What is the difference between p-feature and feature? Thank you!

**Comment by pandanista** [comment]
Let's put this issue into prioritized backlog.

**Comment by pandanista** [comment]
- We are ready to test Update Wiki Page template using RP025.

**Comment by pklmwan** [comment]
Hi @pandanista, I am testing the issue for RP019 [TWE: IS[2022]: RP[Replace 019]: Research: Update Wiki Page]
#721. When testing the template to update the date for RP019, I found that the RP019 folder is empty, so I cannot find any date information. How should I solve this, or should I leave it as is?

**Comment by jinyan0425** [comment]
<img width="1121" alt="Image" src="https://github.com/user-attachments/assets/9b70e2b1-3b10-42de-b2a9-abfc5c6dd18a" />
@pklmwan I got an email about this comment (see the screenshot) but I do not see this here. Did you delete it?

**Comment by pklmwan** [comment]
@jinyan0425 yes, I just write the same comment on the test issue https://github.com/hackforla/internship/issues/721 can you see that?

**Comment by pandanista** [comment]
@pklmwan 

- I am not sure why you are testing the update Wiki template using RP019. Is there a particular reason you chose RP019? The reason I ask is because we mentioned RP025 in this issue. 
    - Did you read the comment [here](https://github.com/hackforla/internship/issues/611#issuecomment-2620164317) or notice the dependency in [the top comment](https://github.com/hackforla/internship/issues/611#issue-2461928724)?

**Comment by jinyan0425** [comment]
> [@pklmwan](https://github.com/pklmwan)
> 
> * I am not sure why you are testing the update Wiki template using RP019. Is there a particular reason you chose RP019? The reason I ask is because we mentioned RP025 in this issue.
>   
>   * Did you read the comment [here](https://github.com/hackforla/internship/issues/611#issuecomment-2620164317) or notice the dependency in [the top comment](https://github.com/hackforla/internship/issues/611#issue-2461928724)?

I left a similar comment in #721. Shall we discuss this here or in #721? @pandanista

**Comment by pandanista** [comment]
I just saw @jinyan0425's comment in issue #721.

@pklmwan Please create a new issue and use RP025 to do the test. Thank you very much.

@jinyan0425 and @pklmwan We can just delete issue #721 so we can start afresh.

**Comment by pandanista** [comment]
> > [@pklmwan](https://github.com/pklmwan)
> > 
> > * I am not sure why you are testing the update Wiki template using RP019. Is there a particular reason you chose RP019? The reason I ask is because we mentioned RP025 in this issue.
> >   
> >   * Did you read the comment [here](https://github.com/hackforla/internship/issues/611#issuecomment-2620164317) or notice the dependency in [the top comment](https://github.com/hackforla/internship/issues/611#issue-2461928724)?
> 
> I left a similar comment in [#721](https://github.com/hackforla/internship/issues/721). Shall we discuss this here or in [#721](https://github.com/hackforla/internship/issues/721)? [@pandanista](https://github.com/pandanista)

I think it is easier to use this issue as it is set up for testing purpose. I am thinking we can delete 721 since it is harder to edit everything there. Your thoughts?

**Comment by jinyan0425** [comment]
@pklmwan We need to test this issue using RP025, as the testing will also help us update the RP025 WIKI, which has recently been updated with new research plan and etc.

Yes, please delete #721, as Ying suggested, and restart the testing using RP025.

@pandanista

**Comment by pandanista** [comment]
@pklmwan and @jinyan0425 I just deleted issue #721 because it is much easier to start afresh. Thank you for understanding!  :)

@pklmwan Another note on the testing: We don't need to keep the `[]`. Technically in any template, anything within the `[]` and including the `[]` should be replaced by the scenario itself, for example RP025.

**Comment by pklmwan** [comment]
@pandanista Thanks I see， I will remake it

**Comment by pandanista** [comment]
- #593 is step 1 and step 2b, the template draft
- This issue is step 2a, the test set up
- #722 is step 3, the test

Once #722 is done, we can deploy the template and close the template issues.
